src/redis-cli.c

```diff
@@ -2131,14 +2182,7 @@ static void repl(void) {
             /* Won't save auth or acl setuser commands in history file */
             int dangerous = 0;
             if (argv && argc > 0) {
```

Comment:
Not necessarily a criticism of this PR, but this code is so weird. We do a null check here, but we error out later if this is NULL, so why don't we just error out earlier? argc == 0 is also silently dropped, but I imagine that is only an error condition.

I bring this up because I would expect we could just call 
```
if (!isDangerousCommand(argc - skipargs, arv + skipargs) {
 // record line
}
```
 and avoid initializing the variable, but we can't because we haven't done the NULL check yet. Not something you need to address, but something I wanted to bring up.

Reply:
I do the refactor in a separate commit. There are some behavior changes.
Previously invalid arguments line or blank line will be log in history file.
I think they should not be log in history file, WDYT?